test/python/circuit/test_gate_power.py

```diff
@@ -0,0 +1,257 @@
+
+        result = SGate().power(1 / 2)
+
+        self.assertEqual(result.label, 's^0.5')
+        self.assertEqual(len(result.definition), 1)
+        self.assertIsInstance(result, Gate)
+        assert_allclose(result.definition[0][0].to_matrix(), expected)
+
+
+@ddt
+class TestGateFloat(QiskitTestCase):
+    """Test power generalization to root calculation"""
+
+    @data(2, 3, 4, 5, 6, 7, 8, 9)
+    def test_direct_root(self, degree):
+        """Test nth root"""
+        result = SGate().power(1 / degree)
+
+        self.assertEqual(result.label, 's^' + str(1 / degree))
+        self.assertEqual(len(result.definition), 1)
+        self.assertIsInstance(result, Gate)
+        assert_allclose(matrix_power(result.definition[0][0].to_matrix(), degree),
+                        SGate().to_matrix())
+
+    @data(2.1, 3.2, 4.3, 5.4, 6.5, 7.6, 8.7, 9.8)
+    def test_float_gt_one(self, exponent):
+        """Test greater-than-one exponents """
+        result = SGate().power(exponent)
+
+        self.assertEqual(result.label, 's^' + str(exponent))
+        self.assertEqual(len(result.definition), 1)
+        self.assertIsInstance(result, Gate)
+        # SGate().to_matrix() is diagonal so `**` is equivalent.
+        assert_allclose(SGate().to_matrix() ** exponent, result.definition[0][0].to_matrix())
+
+    def test_zero_two(self, exponent=0.2):
+        """Test Sgate^(0.2)"""
+        result = SGate().power(exponent)
+
+        self.assertEqual(result.label, 's^' + str(exponent))
+        self.assertEqual(len(result.definition), 1)
+        self.assertIsInstance(result, Gate)
+        assert_allclose(array([[1, 0],
+                               [0, 0.95105652 + 0.30901699j]], dtype=complex),
+                        result.definition[0][0].to_matrix())
+
+    def test_minus_zero_two(self, exponent=-0.2):
+        """Test Sgate^(-0.2)"""
+        result = SGate().power(exponent)
+
+        self.assertEqual(result.label, 's^' + str(exponent))
+        self.assertEqual(len(result.definition), 1)
+        self.assertIsInstance(result, Gate)
+        assert_allclose(array([[1, 0],
+                               [0, 0.95105652 - 0.30901699j]], dtype=complex),
+                        result.definition[0][0].to_matrix())
+
+
+@ddt
+class TestPowerInvariant(QiskitTestCase):
+    """Test Gate.power invariants"""
+
+    @data(-3, -2, -1, 1, 2, 3)
+    def test_invariant1_int(self, n):
+        """Test (op^(1/n))^(n) == op, integer n
+        """
+        qr = QuantumRegister(1, 'qr')
+        circuit = QuantumCircuit(qr)
+        circuit.append(SGate().power(1 / n).power(n), [qr[0]])
+        result = PassManager([Unroller('u3'), Optimize1qGates()]).run(circuit)
+
+        expected_circuit = QuantumCircuit(qr)
+        expected_circuit.append(SGate(), [qr[0]])
+        expected = PassManager([Unroller('u3'), Optimize1qGates()]).run(expected_circuit)
+
+        self.assertEqual(result, expected)
```

Comment:
why compare circuits here and go through transpile? can just check the `gate.to_matrix()` like every other test here.

Reply:
probably is a left over from a different iteration on the feature. Fixed in 7cf5d61f